File: Leetcode/35/Binary_Search_Algo/main.ts
function searchInsert(nums: number[], target: number): number {
    let start: number = 0;
    let end: number = nums.length - 1;

    while(start <= end) {
        // let mid: number = Math.floor(start + (end - start) / 2);
        let mid: number = Math.floor((end + start) / 2);

        if (target < nums[mid]) {
            end = mid - 1; 
        } else if (target > nums[mid]) {
            start = mid + 1;
        } else {
            return mid;
        }
    }

    return start;
};

File: Leetcode/35/Binary_Search_Algo/main.c
int searchInsert(int* nums, int numsSize, int target) {
    int start = 0;
    --numsSize;
    
    while (start <= numsSize) {
        int mid = start + (numsSize - start) / 2;
        // int mid = (numsSize + start) / 2;

        if (target < nums[mid]) {
            numsSize = mid - 1;
        } else if (target > nums[mid]) {
            start = mid + 1;
        } else {
            return mid;
        }
    }

    return start;

    
}